core/src/main/scala/org/apache/spark/rdd/RDDCheckpointData.scala

```diff
@@ -66,6 +88,27 @@ private[spark] class RDDCheckpointData[T: ClassTag](@transient rdd: RDD[T])
     cpFile
   }
 
+  // Get the iterator used to write checkpoint data to HDFS
+  def getCheckpointIterator(
+      rddIterator: Iterator[T],
+      context: TaskContext,
+      partitionId: Int): Iterator[T] = {
+    RDDCheckpointData.synchronized {
+      if (cpState == Initialized) {
+        // Create the output path for the checkpoint
+        val path = new Path(checkpointDir.get, "rdd-" + rddId)
```

Comment:
actually, the path here should just be `checkpointPath`. Right now this duplicates some code.